--- X11-driver-firegl-8.27.10/common/lib/modules/fglrx/build_mod/firegl_public.c.diff_VM_SHM	2006-12-10 11:44:09.000000000 +0100
+++ X11-driver-firegl-8.27.10/common/lib/modules/fglrx/build_mod/firegl_public.c	2006-12-10 11:45:59.000000000 +0100
@@ -27,6 +27,8 @@
 #include <linux/version.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
 #include <linux/utsrelease.h>
+  /* VM_SHM is gone since 2.6.18 http://forums.gentoo.org/viewtopic-t-517697.html?sid=b791467f487342ca3fe1dc887e30d3e9  */
+#define VM_SHM             0x00000000      /* Means nothing: delete it later */
 #endif
 #ifdef MODVERSIONS
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,71)
