--- X11-driver-firegl-8.16.20/lib/modules/fglrx/build_mod/firegl_public.c.orig	2005-11-02 20:29:43.535048712 -0800
+++ X11-driver-firegl-8.16.20/lib/modules/fglrx/build_mod/firegl_public.c	2005-11-02 20:27:19.013019400 -0800
@@ -1475,7 +1485,7 @@ int ATI_API_CALL __ke_copy_to_user(void*
 
 int ATI_API_CALL __ke_verify_area(int type, const void * addr, unsigned long size)
 {
-    return verify_area(type, addr, size);
+    return (access_ok(type, addr, size) ? 0 : -EFAULT);
 }
 
 int ATI_API_CALL __ke_get_pci_device_info(__ke_pci_dev_t* dev, __ke_pci_device_info_t *pinfo)
