--- X11-driver-firegl-8.23.7/common/lib/modules/fglrx/build_mod/firegl_public.c.orig	2006-04-15 02:25:12.183912250 +0200
+++ X11-driver-firegl-8.23.7/common/lib/modules/fglrx/build_mod/firegl_public.c	2006-04-15 02:25:28.484931000 +0200
@@ -21,6 +21,8 @@
 !!! since it requires changes to linux/init/main.c.
 #endif /* !MODULE */
 
+#define __AGP__ 1
+
 // ============================================================
 #include <linux/version.h>
 #ifdef MODVERSIONS
