diff -burN X11-driver-firegl-8.14.13.orig/lib/modules/fglrx/build_mod/agpgart_be.c X11-driver-firegl-8.14.13/lib/modules/fglrx/build_mod/agpgart_be.c
--- X11-driver-firegl-8.14.13.orig/lib/modules/fglrx/build_mod/agpgart_be.c	2005-07-27 13:16:10.000000000 +0000
+++ X11-driver-firegl-8.14.13/lib/modules/fglrx/build_mod/agpgart_be.c	2005-07-27 13:19:22.000000000 +0000
@@ -7067,7 +7067,7 @@
 			if ( agp_bridge.capndx )
             {
 				printk(KERN_INFO PFX "Detected HP ZX1 AGP "
-				       "chipset at %s\n", dev->slot_name);
+				       "chipset at %s\n", pci_name(dev));
 				agp_bridge.type = HP_ZX1;
 				agp_bridge.dev = dev;
               	/* fill in the mode register */
@@ -7133,7 +7133,7 @@
                 if( cap_ptr )
                 {
     				printk(KERN_INFO PFX "Detected a Via Apollo KT400 chipset in AGP v3 mode "
-	    			       "at %s\n", dev->slot_name);
+	    			       "at %s\n", pci_name(dev));
 
                     agp_bridge.type = VIA_APOLLO_KT400;
     				agp_bridge.dev = dev;
@@ -7153,12 +7153,12 @@
                 if( cap_ptr )
                 {
     				printk(KERN_DEBUG PFX "Detected a Via Apollo KT400 chipset in AGP v2 compat mode "
-	    			       "at %s\n", dev->slot_name);
+	    			       "at %s\n", pci_name(dev));
                 }
                 else
                 {
     				printk(KERN_ERR PFX "Detected a Via Apollo KT400 chipset but no AGPv2/v3 caps found "
-	    			       "at %s\n", dev->slot_name);
+	    			       "at %s\n", pci_name(dev));
                 }
             }
             break;
@@ -7184,7 +7184,7 @@
                 if( cap_ptr )
                 {
                     printk(KERN_INFO PFX "Detected a Via P4X400 chipset in AGP v3 mode "
-                           "at %s\n", dev->slot_name);
+                           "at %s\n", pci_name(dev));
 
                     agp_bridge.type = VIA_APOLLO_P4X400;
                     agp_bridge.dev = dev;
@@ -7204,7 +7204,7 @@
                 if( cap_ptr )
                 {
                     printk(KERN_INFO PFX "Detected a Via P4X400 chipset with AGP v3.x caps in AGP v2 compatibility mode "
-                           "at %s\n", dev->slot_name);
+                           "at %s\n", pci_name(dev));
 
                     agp_bridge.type = VIA_APOLLO_P4X400;
                     agp_bridge.dev = dev;
@@ -7224,12 +7224,12 @@
                 if( cap_ptr )
                 {
                     printk(KERN_DEBUG PFX "Detected a Via P4X400 chipset with AGP v2 caps in true AGP v2 mode "
-                           "at %s\n", dev->slot_name);
+                           "at %s\n", pci_name(dev));
                 }
                 else
                 {
                     printk(KERN_ERR PFX "Detected a Via P4X400 chipset but no AGPv2/v3 caps found "
-                           "at %s\n", dev->slot_name);
+                           "at %s\n", pci_name(dev));
                 }
             }
             break;
