diff -urN X11-driver-firegl-8.8.25/lib/modules/fglrx/build_mod/agpgart_be.c X11-driver-firegl-8.8.25.patch/lib/modules/fglrx/build_mod/agpgart_be.c
--- X11-driver-firegl-8.8.25/lib/modules/fglrx/build_mod/agpgart_be.c	2005-01-20 01:46:17.000000000 +0100
+++ X11-driver-firegl-8.8.25.patch/lib/modules/fglrx/build_mod/agpgart_be.c	2005-01-20 01:48:32.000000000 +0100
@@ -95,6 +95,9 @@
 #include <linux/config.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+#define pci_find_class pci_get_class
+#endif /* LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9) */
 
 
 #ifdef MODVERSIONS
